### PR TITLE
:woman_technologist: Min Heap --- Constructor & `size`

### PR DESCRIPTION
**Pull request by jameshughes89**
### What
Start the min heap class. Start with constructor and size method. 

### Testing
:+1:

### Additional Notes
1. I thought I had actually implemented this for cs102 since there was a lab on the topic, but I guess not. 
2. I was considering having a more general `ArrayHeap` class that takes a `Comparator`, but I think that's going to be a little much.

### CODE REVIEW

File: src/main/java/ArrayMinHeap.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+    private static final int DEFAULT_CAPACITY = 100;
+
+    private int size;
+    private T[] heap;
+
+    /**
+     * Create a new empty heap with the array container being created with the default capacity.
+     */
+    public ArrayMinHeap(){
```
Comment (twentylemon) [inline src/main/java/ArrayMinHeap.java]: intellij has a "format on save" option. use it.

File: src/main/java/Heap.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
+public interface Heap <T extends Comparable<? super T>> extends Iterable<T> {
+
+    /**
+     * Add the provided element to the heap such that the heap ordering is preserved.
+     *
+     * @param element Element to be added to the heap.
+     */
+    public void add(T element);
```
Comment (twentylemon) [inline src/main/java/Heap.java]: Similar to #11, return `boolean`.

File: src/main/java/Heap.java [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+    /**
+     * Remove the element at the top of the heap. This may be the min or max element depending on the specific heap
+     * implementation.
+     *
+     * @return Element at the top of the heap.
+     */
+    public T remove();
```
Comment (twentylemon) [inline src/main/java/Heap.java]: Interfaces don't need `public` on their methods.

**Review by twentylemon — APPROVED** [review]

File: src/main/java/ArrayMinHeap.java (hunk 1 above)
Comment (jameshughes89) [reply src/main/java/ArrayMinHeap.java]: It seems to not be on by default with "new" projects.

**Comment by jameshughes89** [comment]
added an `isEmpty` method to the `Heap` since it would be used for a priority queue of some sort, and that method exists for the queues we've made so far.

**Review by twentylemon — APPROVED** [review]

**Pull request merged**